graal-js/src/com.oracle.truffle.js/src/com/oracle/truffle/js/runtime/builtins/JSTemporalPlainDate.java

```diff
@@ -0,0 +1,265 @@
+        PrototypeSupplier{
+
+    public static final JSTemporalPlainDate INSTANCE = new JSTemporalPlainDate();
+
+    public static final String CLASS_NAME = "TemporalPlainDate";
+    public static final String PROTOTYPE_NAME = "TemporalPlainDate.prototype";
+
+    public static final String YEAR = "year";
+    public static final String MONTH = "month";
+    public static final String DAY = "day";
+
+    private JSTemporalPlainDate() {
+    }
+
+    @Override
+    public String getClassName(DynamicObject object) {
+        return "Temporal.PlainDate";
+    }
+
+    @Override
+    public String getClassName() {
+        return CLASS_NAME;
+    }
+
+    @Override
+    public DynamicObject createPrototype(JSRealm realm, DynamicObject constructor) {
+        JSContext ctx = realm.getContext();
+        DynamicObject prototype = JSObjectUtil.createOrdinaryPrototypeObject(realm);
+
+        JSObjectUtil.putToStringTag(prototype, "Temporal.PlainDate");
+
+        return prototype;
+    }
+
+    @Override
+    public Shape makeInitialShape(JSContext context, DynamicObject prototype) {
+        Shape initialShape = JSObjectUtil.getProtoChildShape(prototype, JSTemporalPlainDate.INSTANCE, context);
+        return initialShape;
+    }
+
+    @Override
+    public DynamicObject getIntrinsicDefaultProto(JSRealm realm) {
+        return realm.getTemporalPlainDatePrototype();
+    }
+
+    public static JSConstructor createConstructor(JSRealm realm) {
+        return INSTANCE.createConstructorAndPrototype(realm);
+    }
+
+    public static boolean isJSTemporalPlainDate(Object obj) {
+        return obj instanceof JSTemporalPlainDateObject;
+    }
+
+    public static DynamicObject createTemporalDate(JSContext context, long y, long m, long d) {
+        rejectDate(y, m, d);
+        if(!dateTimeWithinLimits(y, m, d, 12, 0, 0, 0, 0, 0)) {
+            throw Errors.createRangeError("Date is not within range.");
+        }
+        // TODO: Check if calendar is not an object.
+        JSRealm realm = context.getRealm();
+        JSObjectFactory factory = context.getTemporalPlainDateFactory();
+        DynamicObject object = factory.initProto(new JSTemporalPlainDateObject(factory.getShape(realm),
+                (int) y, (int)m, (int)d
+        ), realm);
+        return context.trackAllocation(object);
+    }
+
+    private static void rejectDate(long year, long month, long day) {
+        if (!validateDate(year, month, day)) {
+            throw Errors.createRangeError("Given date outside the range.");
+        }
+    }
+
+    private static boolean validateDate(long year, long month, long day) {
+        if (month < 1 || month > 12) {
+            return false;
+        }
+        long daysInMonth = daysInMonth(year, month);
+        if (day < 1 || day > daysInMonth) {
+            return false;
+        }
+        return true;
+    }
+
+    private static int daysInMonth(long year, long month) {
```

Comment:
Not sure about the details here, but my gutt feeling tells me methods like those should be in `TemporalUtil`  as they might be reusable in other parts later?